src/core/toga/sources/row.py

```diff
@@ -0,0 +1,50 @@
+
+class Row:
+    def __init__(self, **data):
+        self._attrs = list(data.keys())
+        self._source = None
+        for name, value in data.items():
+            setattr(self, name, value)
+
+    ######################################################################
+    # Utility wrappers
+    ######################################################################
+
+    def __setattr__(self, attr, value):
+        super().__setattr__(attr, value)
+        if attr in self._attrs and self._source is not None:
+            self._source._notify('change', item=self)
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.as_dict() == other.as_dict()
+
+    def __repr__(self):
+        return repr(self.as_dict())
+
+    def as_dict(self):
+        return {attr: getattr(self, attr) for attr in self._attrs}
+
+    @classmethod
+    def create_row(cls, data, accessors, source=None):
```

Comment:
I'm interested in your reasoning on this change. The original version had this as a method on Source, which meant a subclass could choose to use an internal representation other than Row if that was convenient. Moving it to Row isn't inherently *wrong*, but I'm wondering if there was a conscious decision process around it.

Reply:
There's also an interesting discrepancy with TreeSource if create_row is moved to Row, rather than being on a Source base class.

Reply:
I made this change for 2 main reasons:
1. Using a class method to create an instance of the class is a quite common design pattern. It enforces single responsibility and preserves encapsulation.
2. `create_row` was a method of `ListSource`, not `Source`. I wanted `StackSource` and `ListSource` to be distinct from each other, but that both would use `Row`. Therefore, I moved the method from there.

A solution might be to move the method to `CollectionSource`, which is a new interface I introduced for all those sources to inherit from.

Let me know what you prefer, and I'll make the change :)

Reply:
I think moving to CollectionSource would be my preferred option.